util/json.go

```diff
@@ -114,6 +114,10 @@ func Reference(x interface{}) *interface{} {
 
 // Unmarshal decodes a YAML, JSON or JSON extension value into the specified type.
 func Unmarshal(bs []byte, v interface{}) error {
+	if bs[0] == 0xef && bs[1] == 0xbb && bs[2] == 0xbf {
+		bs = bs[3:] // Strip UTF-8 BOM
+	}
```

Comment:
Maybe link the RFC for context.

Reply:
Added! Also added a `len` check on `bs` to avoid empty input leading to out of bounds refs... the tests didn't like those, hehe